marimo/_runtime/runner/hooks_post_execution.py

```diff
@@ -165,14 +167,21 @@ def _broadcast_data_source_connection(
         ).broadcast()
 
     for variable, engine in engines:
-        tables = engine.get_tables()
-        if tables:
-            LOGGER.debug(f"Broadcasting engine tables for {variable}")
-            Datasets(tables=tables).broadcast()
+        if isinstance(engine, SQLAlchemyEngine):
+            databases = engine.get_databases(
+                include_schemas=True,
+                include_tables=True,
+                include_table_details=False,
+            )
+        elif isinstance(engine, DuckDBEngine):
+            databases = engine.get_databases()
+        if databases:
```

Comment:
`databases` could be unbound if not of the engines exist. so I would declare 
 `databases: Optional[Databases]` on line 170